<?xml version="1.0"?>
<launch>
    <!-- Arguments -->
    <arg name="preview_resolution" default="1080" doc="Preview resolution: 4k, 2k, 1080, 960"/>
    <arg name="recording_resolution" default="8k" doc="SD recording resolution: 8k, 5.7k"/>
    <arg name="auto_start_recording" default="true" doc="Start SD recording on launch"/>
    <arg name="battery_poll_interval_sec" default="30" doc="Battery check interval in seconds"/>
    <arg name="low_battery_threshold_percent" default="15" doc="Auto-save when battery below this %"/>
    <arg name="publish_combined" default="true" doc="Publish combined dual-fisheye image"/>
    <arg name="combined_format" default="sidebyside" doc="Combined image format: sidebyside or topbottom"/>
    <arg name="use_gps_time" default="true" doc="Use GPS time from external topic"/>
    <arg name="gps_time_topic" default="/ext/time" doc="Topic providing GPS time reference"/>

    <!-- X5 Driver Node -->
    <node name="x5_driver" pkg="x5_ros_driver" type="x5_driver_node" output="screen" required="true">
        <!-- Load default params -->
        <rosparam file="$(find x5_ros_driver)/config/x5_params.yaml"/>
        
        <!-- Override with launch args -->
        <param name="preview_resolution" value="$(arg preview_resolution)"/>
        <param name="recording_resolution" value="$(arg recording_resolution)"/>
        <param name="auto_start_recording" value="$(arg auto_start_recording)"/>
        <param name="battery_poll_interval_sec" value="$(arg battery_poll_interval_sec)"/>
        <param name="low_battery_threshold_percent" value="$(arg low_battery_threshold_percent)"/>
        <param name="publish_combined" value="$(arg publish_combined)"/>
        <param name="combined_format" value="$(arg combined_format)"/>
        <param name="use_gps_time" value="$(arg use_gps_time)"/>
        <param name="gps_time_topic" value="$(arg gps_time_topic)"/>
    </node>
</launch>
